houston/recorder.py

```diff
@@ -0,0 +1,75 @@
+__all__ = ['Recorder']
+
+from typing import Type, FrozenSet, Dict, Any, List
+import attr
+import logging
+
+import json
+import threading
+
+from .state import State
+
+logger = logging.getLogger(__name__)  # type: logging.Logger
+logger.setLevel(logging.DEBUG)
+
+
+class NoFileSetError(BaseException):
```

Comment:
This should inherit from a `HoustonException` class (if we don't have one yet, we should add one). In any case, avoid subclassing `BaseException`: `BaseException` is used by _all_ exceptions, including those that are effectively used as signals (e.g., `KeyboardInterrupt`). Using `BaseException` as the base class will prevent typical "catch-all" exception handlers from working.